Production/plugins/TauTupleProducer.cc

```diff
@@ -66,86 +66,33 @@ struct TauTupleProducerData {
     using clock = std::chrono::system_clock;
 
     const clock::time_point start;
-    tau_tuple::TauTuple tauTuple;
-    tau_tuple::SummaryTuple summaryTuple;
-    std::mutex mutex;
+    std::unique_ptr<tau_tuple::TauTuple> tauTuple;
+    std::unique_ptr<tau_tuple::SummaryTuple> summaryTuple;
 
-private:
-    size_t n_producers;
-
-    TauTupleProducerData(TFile& file) :
-        start(clock::now()),
-        tauTuple("taus", &file, false),
-        summaryTuple("summary", &file, false),
-        n_producers(0)
-    {
-        summaryTuple().numberOfProcessedEvents = 0;
-    }
-
-    ~TauTupleProducerData() {}
-
-public:
-
-    static TauTupleProducerData* RequestGlobalData()
```

Comment:
Just for my education, why the methods `*GlobalData` were used before? (and not needed now)

Reply:
It comes from a custom legacy code. It was an attempt to improve performance by parallelising producer, but we end up having `lock(mutex)` in the beginning of `produce` method, so it never really worked. If in the future we decide to implement multi-threading support during tuple production, we can use standard CMSSW interfaces for this.